policyengine_api/services/policy_service.py

```diff
@@ -18,3 +59,134 @@ def get_policy_json(self, country_id, policy_id):
         except Exception as e:
             print(f"Error getting policy json: {str(e)}")
             raise e
+
+    def set_policy(
+        self, country_id: str, label: str, policy_json: dict
+    ) -> tuple[int, str, bool]:
+        """
+        Insert a new policy into the database
+
+        Arguments
+            country_id: str -- the country ID
+            label: str -- the policy label
+            policy_json: dict -- the policy content
+
+        Returns
+            tuple[int, str, bool] -- the new policy ID, a message, and whether or not
+            the policy already existed
+        """
+        print("Setting new policy")
+
+        try:
+
+            policy_hash = hash_object(policy_json)
+            api_version = COUNTRY_PACKAGE_VERSIONS.get(country_id)
+            # Check if policy already exists
+            print("Checking if policy exists")
+            existing_policy = self._get_unique_policy_with_label(
+                country_id, policy_hash, label
+            )
+
+            # If so, pass appropriate values back
+            if existing_policy:
+                print("Policy already exists")
+                existing_policy_id = str(existing_policy["id"])
+                message = (
+                    "Warning: Record created previously with this label. To create "
+                    + "a new record, change the submitted data's country ID, policy "
+                    + "parameters, or label, and emit the request again"
+                )
+                return existing_policy["id"], "Policy already exists", True
+
+            # Otherwise, insert the new policy...
+            print("Policy does not exist; creating new policy")
+            self._create_new_policy(
+                country_id, policy_json, policy_hash, label, api_version
+            )
+
+            # And then fetch it back out; once an ORM is added, we can
+            # just return policy ID directly from the insert
+            new_policy = self._get_unique_policy_with_label(
+                country_id, policy_hash, label
+            )
+
+            return int(new_policy["id"]), "Policy created", False
+
+        except Exception as e:
+            print(f"Error setting policy: {str(e)}")
+            raise e
+
+    def _create_new_policy(
+        self,
+        country_id: str,
+        policy_json: dict,
+        policy_hash: str,
+        label: str | None,
+        api_version: str,
+    ) -> None:
+        """
+        Insert a new policy into the database
+
+        Arguments
+            country_id: str -- the country ID
+            policy_json: dict -- the policy content
+            policy_hash: str -- the policy content hash
+            label: str | None -- the policy label
+            api_version: str -- the API version
```

Comment:
**very nitty, nonblocking** I really like Clean Code's approach to doc comments (paraphrasing):
1. don't just restate the obvious, it doesn't add value and it creates additional maintenance.
2. if you do need a comment to clarify something, ask yourself if the code could be written to make it obvious.

In this case specifically, we already know the names and types of the method arguments from the code and they're clearly named (I.e. I know country_id is the country Id).  Why incur the additional debt of having to keep the comment and the declaration 100% in sync?

"Insert a new policy into the database" is maybe adding some info, but nothing we couldn't easily capture by calling the method _insert_policy_into_db or whatever.